src/Symfony/Component/Runtime/Internal/SymfonyErrorHandler.php

```diff
@@ -24,12 +24,31 @@ class SymfonyErrorHandler
 {
     public static function register(bool $debug): void
     {
-        BasicErrorHandler::register($debug);
+        if (!class_exists(ErrorHandler::class)) {
+            BasicErrorHandler::register($debug);
 
-        if (class_exists(ErrorHandler::class)) {
+            return;
+        }
+
+        error_reporting(-1);
+
+        if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
+            ini_set('display_errors', $debug);
+        } elseif (!filter_var(\ini_get('log_errors'), \FILTER_VALIDATE_BOOL) || \ini_get('error_log')) {
+            // CLI - display errors only if they're not already logged to STDERR
+            ini_set('display_errors', 1);
+        }
+
+        if (0 <= \ini_get('zend.assertions')) {
+            ini_set('zend.assertions', (int) $debug);
```

Comment:
@nicolas-grekas why are you disabling assertions in non-debug mode ? We are relying on assertions in production  mode and this change breaks our application.